File: Data Structures & Algorithms/binary-search/submission-7.java
class Solution {
    public int search(int[] nums, int target) {
        int base_idx = 0;
        while (nums.length != 0) {
            int mid = nums.length / 2;
            if (nums[mid] == target) {
                return base_idx + mid;
            }
            else if (nums[mid] < target) {
                nums = Arrays.copyOfRange(nums, mid + 1, nums.length);
                base_idx += mid + 1;
            }
            else {
                nums = Arrays.copyOfRange(nums, 0, mid);
                base_idx += 0;
            }
        }
        return -1;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-7.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        my_dict = {}
        for i in range(len(nums)):
            comp = target - nums[i]
            print("comp", comp, "val", nums[i], my_dict)
            if comp in my_dict:
                val = nums[i]
                return [my_dict[comp], i]
            else:
                my_dict[nums[i]] = i